fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/impl/SkolemNodeRdfContext.java

```diff
@@ -70,10 +77,21 @@ public SkolemNodeRdfContext(final FedoraResource resource,
         final Iterator<Property> references = Iterators.filter(properties,
                 uncheck((final Property p) -> REFERENCE_TYPES.contains(p.getType()))::test);
 
-        final Iterator<Node> nodes = Iterators.transform(new PropertyValueIterator(references),
-                UncheckedFunction.uncheck((final Value v) -> v.getType() == PATH ?
-                        session().getNode(v.getString()) : session().getNodeByIdentifier(v.getString()))::apply);
+        final Iterator<Node> nodes = Iterators.transform(new PropertyValueIterator(references), valueToNode::apply);
 
-        return Iterators.filter(nodes, isSkolemNode::test);
+        return Iterators.filter(Iterators.filter(nodes, Objects::nonNull), isSkolemNode::test);
```

Comment:
Yes, there is: https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html#and-java.util.function.Predicate-